admin/admin.py

```diff
@@ -105,15 +125,51 @@ def save_settings(settings: Dict[str, Any], flash_msg: str) -> Response:
 
     :returns: Redirect with flash message
     """
-    config_file_path = path.join(app.config['APP_SHARED_FOLDER'], 'banner_settings.json')
+    config_file_path = path.join(app.config['APP_SHARED_FOLDER'], 'admin_settings.json')
     app.config.update(settings)
 
     try:
-        with open(config_file_path, 'w') as file:
-            json.dump(settings, file)
+        # Load existing settings and merge with new settings
+        with open(config_file_path, 'r+') as file:
+            file_settings = json.load(file)
+            file_settings.update(settings)
+            # Move to the beginning of the file to overwrite
+            file.seek(0)
+            json.dump(file_settings, file, indent=4)
+            file.truncate()
         flash(flash_msg, 'success')
     except Exception:
         flash("Failed to save settings", "danger")
         return "Failed to save settings", 500
 
-    return redirect(url_for('admin_bp.index'))
+    if "YODA_THEME" in settings.keys():
+        # Load the theme template if the current theme is changed
+        set_theme_loader(app, remove_cache=True)
+    return redirect(url_for("admin_bp.index"))
+
+
+def set_theme_loader(app: Flask, remove_cache: Optional[bool] = False) -> None:
+    """
+    Configures the template loader with the updated theme.
+
+    :param app: The Flask application instance to configure.
+    :param remove_cache: A boolean flag indicates whether to clear the template cache. Defaults to False.
+    """
+    # Target theme path
+    theme_path = path.join(app.config.get('YODA_THEME_PATH'), app.config.get('YODA_THEME'))
+    # Secondary theme path for scanning missing templates
+    default_theme_path = "/var/www/yoda/general/templates/general"
```

Comment:
We should introduce a variable for the path `/var/www/yoda/`, which could be defined in the Flask configuration file. Additionally, there are two other instances where hardcoded paths are used, and these can also be updated to use this variable. This modification would allow for the flexibility to install the portal in a different directory if needed.

Reply:
Leave for another task. 
See the quote from Lazlo: "I think that would be an overall improvement. The comment was not specific to your changeset, but more a general improvement we should do imho.
 
We can leave it as is for this PR and pick it up later as a separate task."